ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -240,46 +200,65 @@ def leak_message_located(
                         max_width,
                     )
                     leak_msg.write(formatted_line)
+                else:
+                    # The secret is on multiple lines
+                    detector_position = float("inf"), float("-inf")
+                    for index, line_of_secret in enumerate(match.lines_with_secret):
+                        leak_msg.write(
+                            line_of_secret.build_line_count(padding, is_secret=True)
+                        )
+                        secret_start = span.column_index_start if index == 0 else 0
+                        secret_end = (
+                            span.column_index_end
+                            if index == len(match.lines_with_secret) - 1
+                            else len(line_of_secret.content)
+                        )
+                        formatted_line, secret_position = format_line_with_secret(
+                            line_of_secret.content, secret_start, secret_end, max_width
+                        )
+                        leak_msg.write(formatted_line)
 
+                        detector_position = (
+                            min(detector_position[0], secret_position[0]),
+                            max(detector_position[1], secret_position[1]),
+                        )
+                    line_number += len(match.lines_with_secret) - 1
                 detector_position = int(detector_position[0]), int(detector_position[1])
-                detector = format_detector(flat_match.match_type, *detector_position)
+                detector = format_detector(match.match_type, *detector_position)
                 leak_msg.write(display_detector(detector, offset))
 
-        # The current line is just here for context
-        else:
-            leak_msg.write(line.build_line_count(padding, is_secret=False))
-            if clip_long_lines:
-                line_content = clip_long_line(line_content, max_width, after=True)
-            leak_msg.write(f"{display_patch(line_content)}\n")
-
-        old_line_number = multiline_end if multiline_end else line_number
+        old_line_number = line_number
 
     return leak_msg.getvalue()
 
 
 def flatten_policy_breaks_by_line(
     policy_breaks: List[PolicyBreak],
-) -> Dict[int, List[ExtendedMatch]]:
+) -> OrderedDict[Line, List[ExtendedMatch]]:
     """
     flatten_policy_breaks_by_line turns a list of policy breaks into a dictionary
     mapping a line number to a list of matches starting at that line.
```

Comment:
docstring needs to be updated: the returned dict keys are not line numbers anymore.